src/vis/dataviews-tracker.js

```diff
@@ -0,0 +1,64 @@
+var Backbone = require('backbone');
+
+var DataviewsChangeTracker = Backbone.Collection.extend({
+  initialize: function (models, options) {
+    this.vis = options.vis;
+    this._tracking = false;
+
+    this._initBinds();
+  },
+
+  _initBinds: function () {
+    this.on('add', this._patch, this);
+  },
+
+  track: function () {
+    this._tracking = true;
+    this._check();
+  },
+
+  _patch: function (dataviewModel) {
```

Comment:
Why not let each dataview keep track of its status [in its own fetch method](https://github.com/CartoDB/cartodb.js/blob/v4/src/dataviews/dataview-model-base.js#L253)?

```

var UNFETCHED_STATUS = 'unfeteched';
var FETCHING_STATUS = 'fetching';
var FETCHED_STATUS = 'fetched;
var FETCH_ERROR_STATUS = 'fetched;

var DataviewModelBase = Model.extend({
  default: {
    ...,
    status: UNFETCHED_STATUS,
  },

  ...,

  fetch: function (options) {
    this.set('status', FETCHING_STATUS);

    ...

    return Model.prototype.fetch.call(this, _.extend(opts, {
      success: function () {
        
        ...
        this.set('status', FETCHED_STATUS);
      }.bind(this),
      error: function (mdl, err) {
        ...
        this.set('status', FETCH_ERROR_STATUS)
      }.bind(this)
    }));
  },

  isFetched: function () {
    return this.get('status', FETCHED_STATUS);
  },

  ...
});
```